openapi/weaviate/utils.bal

```diff
@@ -0,0 +1,214 @@
+isolated function getSerializedArray(string arrayName, anydata[] anyArray, string style = "form", boolean explode = true) returns string {
+    string key = arrayName;
+    string[] arrayValues = [];
+    if (anyArray.length() > 0) {
+        if (style == FORM && !explode) {
+            arrayValues.push(key, "=");
+            foreach anydata i in anyArray {
+                arrayValues.push(getEncodedUri(i.toString()), ",");
+            }
+        } else if (style == SPACEDELIMITED && !explode) {
+            arrayValues.push(key, "=");
+            foreach anydata i in anyArray {
+                arrayValues.push(getEncodedUri(i.toString()), "%20");
+            }
+        } else if (style == PIPEDELIMITED && !explode) {
+            arrayValues.push(key, "=");
+            foreach anydata i in anyArray {
+                arrayValues.push(getEncodedUri(i.toString()), "|");
+            }
+        } else if (style == DEEPOBJECT) {
+            foreach anydata i in anyArray {
+                arrayValues.push(key, "[]", "=", getEncodedUri(i.toString()), "&");
+            }
+        } else {
+            foreach anydata i in anyArray {
+                arrayValues.push(key, "=", getEncodedUri(i.toString()), "&");
+            }
+        }
+        _ = arrayValues.pop();
+    }
+    return string:'join("", ...arrayValues);
+}
+
+# Serialize the array of records according to the form style.
+#
+# + parent - Parent record name
+# + value - Array of records to be serialized
+# + style - Defines how multiple values are delimited
+# + explode - Specifies whether arrays and objects should generate separate parameters
+# + return - Serialized record as a string
+isolated function getSerializedRecordArray(string parent, record {}[] value, string style = FORM, boolean explode = true) returns string {
+    string[] serializedArray = [];
+    if style == DEEPOBJECT {
+        int arayIndex = 0;
+        foreach var recordItem in value {
+            serializedArray.push(getDeepObjectStyleRequest(parent + "[" + arayIndex.toString() + "]", recordItem), "&");
+            arayIndex = arayIndex + 1;
+        }
+    } else {
+        if (!explode) {
+            serializedArray.push(parent, "=");
+        }
+        foreach var recordItem in value {
+            serializedArray.push(getFormStyleRequest(parent, recordItem, explode), ",");
+        }
+    }
+    _ = serializedArray.pop();
+    return string:'join("", ...serializedArray);
+}
+
+# Get Encoded URI for a given value.
+#
+# + value - Value to be encoded
+# + return - Encoded string
+isolated function getEncodedUri(anydata value) returns string {
+    string|error encoded = url:encode(value.toString(), "UTF8");
+    if (encoded is string) {
+        return encoded;
+    } else {
+        return value.toString();
+    }
+}
+
+# Generate query path with query parameter.
+#
+# + queryParam - Query parameter map
+# + encodingMap - Details on serialization mechanism
```

Comment:
```suggestion
# + encodingMap - Details of the serialization mechanism
```

Reply:
Since the `utils.bal` file is generated by the tool, let's not change the content.